src/datatypes/temporal_value.c

```diff
@@ -0,0 +1,339 @@
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+	int month = timeDescriptor->tm_mon;
+	return (month / 3) + 1;
+}
+
+int64_t RG_TemporalValue_GetMonth(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+	return timeDescriptor->tm_mon + 1;
+}
+
+int64_t RG_TemporalValue_GetWeek(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+
+	int64_t y = 0, w = 0;
+	int err = _timeDescriptor_YearWeek(timeDescriptor, &y, &w);
+	return w;
+}
+
+int64_t RG_TemporalValue_GetWeekYear(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+
+	int64_t y = 0, w = 0;
+	int err = _timeDescriptor_YearWeek(timeDescriptor, &y, &w);
+	return y;
+}
+
+int64_t RG_TemporalValue_GetDayOfQuarter(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+
+	int q = timeDescriptor->tm_mon / 4;
+	if(isLeapYear(timeDescriptor->tm_year))
+		return timeDescriptor->tm_yday - leapYearQuarterDaysAgg[q] + 1;
+	else
+		return timeDescriptor->tm_yday - normalYearQuarterDaysAgg[q] + 1;
+}
+
+int64_t RG_TemporalValue_GetDay(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+
+	return timeDescriptor->tm_mday;
+}
+
+int64_t RG_TemporalValue_GetOrdinalDay(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+
+	return timeDescriptor->tm_yday + 1;
+}
+
+int64_t RG_TemporalValue_GetDayOfWeek(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & dateMask)) {
+		// not a date type. should return error/execption
+		return INT64_MIN;
+	}
+	struct tm *timeDescriptor  = _getTimeDescriptorFromTemporalValue(temporalValue);
+	// tm first day is Sunday, Cypher first day is Monday
+	return (timeDescriptor->tm_wday + 6) % 7 + 1;
+}
+
+int64_t RG_TemporalValue_GetHour(RG_TemporalValue temporalValue) {
+	if(!(temporalValue.type & timeMask)) {
+		// not a time type. should return error/execption
+		return INT64_MIN;
```

Comment:
Either define a value indicating an error, e.g. #define
or change the signature of this function and all other alike to:
Into RG_TemporalValue_GetHour(RG_TemporalValue temporalValue, int64_t*hour)
hour will be set only if temporalValue represents time and return SUCCESS, otherwise hour is set to NULL or left as is and return an error indicator.

Reply:
done